tests/models/test_prompt_segmentor.py

```diff
@@ -0,0 +1,273 @@
+"""Unit test package for Prompt Segmentor."""
+
+from __future__ import annotations
+
+# ! The garbage collector
+import multiprocessing
+import shutil
+from pathlib import Path
+from typing import Callable
+
+import numpy as np
+import pytest
+
+from tiatoolbox.models import PromptSegmentor
+from tiatoolbox.models.architecture.sam import SAM
+from tiatoolbox.models.engine.semantic_segmentor import (
+    IOSegmentorConfig,
+)
+from tiatoolbox.utils import env_detection as toolbox_env
+from tiatoolbox.utils import imwrite
+from tiatoolbox.utils.misc import select_device
+from tiatoolbox.wsicore.wsireader import WSIReader
+
+ON_GPU = toolbox_env.has_gpu()
+BATCH_SIZE = 1 if not ON_GPU else 2
+try:
+    NUM_LOADER_WORKERS = multiprocessing.cpu_count()
+except NotImplementedError:
+    NUM_LOADER_WORKERS = 2
+
+
+def test_functional_segmentor(
+    remote_sample: Callable,
+    tmp_path: Path,
+) -> None:
+    """Functional test for segmentor."""
+    save_dir = tmp_path / "dump"
+    # # convert to pathlib Path to prevent wsireader complaint
+    resolution = 2.0
+    mini_wsi_svs = Path(remote_sample("patch-extraction-vf"))
+    reader = WSIReader.open(mini_wsi_svs, resolution)
+    thumb = reader.slide_thumbnail(resolution=resolution, units="mpp")
+    thumb = thumb[63:191, 750:878, :]
+    mini_wsi_jpg = f"{tmp_path}/mini_svs.jpg"
+    imwrite(mini_wsi_jpg, thumb)
+
+    # preemptive clean up
+    shutil.rmtree(save_dir, ignore_errors=True)
+
+    model = SAM()
+
+    # test engine setup
+
+    _ = PromptSegmentor(None, BATCH_SIZE, NUM_LOADER_WORKERS)
+
+    prompt_segmentor = PromptSegmentor(model, BATCH_SIZE, NUM_LOADER_WORKERS)
+
+    ioconfig = IOSegmentorConfig(
+        input_resolutions=[
+            {"units": "mpp", "resolution": 4.0},
+        ],
+        output_resolutions=[{"units": "mpp", "resolution": 4.0}],
+        patch_input_shape=[512, 512],
+        patch_output_shape=[512, 512],
+        stride_shape=[512, 512],
+    )
+
+    # test inference
+
+    points = np.array([[[64, 64]], [[64, 64]]])  # Point on nuclei
```

Comment:
why two of the same points? Does it not make more sense to have two different points here?